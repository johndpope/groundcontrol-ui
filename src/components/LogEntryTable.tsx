// Copyright 2019 Stratumn
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import graphql from "babel-plugin-relay/macro";
import React from "react";
import { createFragmentContainer } from "react-relay";
import { Table } from "semantic-ui-react";

import { LogEntryTable_items } from "./__generated__/LogEntryTable_items.graphql";

import { IProps as ILogEntryMessageProps } from "./LogEntryMessage";
import LogEntryTableRow from "./LogEntryTableRow";

export interface IProps {
  items: LogEntryTable_items;
  onClickSourceFile: (values: ILogEntryMessageProps) => any;
}

export function LogEntryTable({ items, onClickSourceFile }: IProps) {
  if (items.length < 1) {
    return (
      <Table inverted={true}>
        <Table.Body>
          <Table.Row>
            <Table.Cell>There are no log entries at this time.</Table.Cell>
          </Table.Row>
        </Table.Body>
      </Table>
    );
  }

  const rows = items.map((item, index) => (
    <LogEntryTableRow
      key={item.id}
      item={item}
      prevItem={index > 0 ? items[index - 1] : null}
      onClickSourceFile={onClickSourceFile}
    />
  ));

  return (
    <Table inverted={true}>
      <Table.Body>{rows}</Table.Body>
    </Table>
  );
}

export default createFragmentContainer(LogEntryTable, graphql`
  fragment LogEntryTable_items on LogEntry
    @relay(plural: true) {
    ...LogEntryTableRow_item
    ...LogEntryTableRow_prevItem
    id
  }`,
);
