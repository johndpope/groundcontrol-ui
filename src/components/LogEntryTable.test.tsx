// Copyright 2019 Stratumn
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { shallow } from "enzyme";
import React from "react";

import { mockQueryPropAttrs } from "../testing/relay";

import { LogEntryTable } from "./LogEntryTable";

const props = {
  items: [
    {
      ...mockQueryPropAttrs(),
      id: "id1"
    },
    {
      ...mockQueryPropAttrs(),
      id: "id2"
    }
  ],
  onClickSourceFile: jest.fn()
};

describe("<LogEntryTable />", () => {
  it("renders items correctly", () => {
    const wrapper = shallow(<LogEntryTable {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly without items", () => {
    const wrapper = shallow(<LogEntryTable {...props} items={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
