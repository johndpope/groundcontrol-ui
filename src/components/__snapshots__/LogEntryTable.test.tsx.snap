// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogEntryTable /> renders correctly without items 1`] = `
<Table
  as="table"
  inverted={true}
>
  <TableBody
    as="tbody"
  >
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableCell
        as="td"
      >
        There are no log entries at this time.
      </TableCell>
    </TableRow>
  </TableBody>
</Table>
`;

exports[`<LogEntryTable /> renders items correctly 1`] = `
<Table
  as="table"
  inverted={true}
>
  <TableBody
    as="tbody"
  >
    <ForwardRef(Relay(LogEntryTableRow))
      item={
        Object {
          " $fragmentRefs": null,
          " $refType": null,
          "id": "id1",
        }
      }
      key="id1"
      onClickSourceFile={[MockFunction]}
      prevItem={null}
    />
    <ForwardRef(Relay(LogEntryTableRow))
      item={
        Object {
          " $fragmentRefs": null,
          " $refType": null,
          "id": "id2",
        }
      }
      key="id2"
      onClickSourceFile={[MockFunction]}
      prevItem={
        Object {
          " $fragmentRefs": null,
          " $refType": null,
          "id": "id1",
        }
      }
    />
  </TableBody>
</Table>
`;
